react/react2/week1/homework/src/App.js

```diff
@@ -1,7 +1,10 @@
 import './App.css';
-import React, {useCallback, useEffect, useState} from 'react';
+import React, { useCallback, useEffect, useState, createContext } from 'react';
+import H1Title from './H1Title'
 
-function App() {
+export const AppContext = createContext(null)
+
+export default function App() {
   const [loading, setLoading] = useState(true)
```

Comment:
You don't need so many states: it's enough one for a search query, one the rest are for fetch status (loading, data, error)
Api address doesn't change, it's a constant data, hence state is not needed